record/record.go

```diff
@@ -71,8 +71,6 @@
 // next block and looks for the next full or first chunk.
 package record // import "github.com/golang/leveldb/record"
 
-// TODO: implement the recovery algorithm.
```

Comment:
Basically, jump to the next 32KB aligned offset, and start scanning for the first FULL or FIRST chunk type.

SkipToInitialOffset is an additional feature from the C++ version, that allows you to use LevelDB as sharded input to a processing system i.e. mapreduce, since you can split a file at arbitrary offsets (32KB aligned of course) and then proceed scanning from there. 

I believe C++ uses this method because the abstracted file object it uses is a SequentialFile object that doesn't support seeking in arbitrary directions, so they implemented it like that.

CurrentOffset was added because the it felt more readable than f.Seek(0, os.SEEK_CUR) for finding the current cursor offset.

Happy to go with whatever approach you think is most idiomatic.